pkg/cloud/cloud.go

```diff
@@ -509,26 +505,17 @@ func (c *cloud) ResizeOrModifyDisk(ctx context.Context, volumeID string, newSize
 		return 0, fmt.Errorf("unable to modify AWS volume %q: %w", volumeID, err)
 	}
 
-	mod := response.VolumeModification
-	state := aws.StringValue(mod.ModificationState)
-
-	if volumeModificationDone(state) {
-		if newSizeBytes != 0 {
-			return c.checkDesiredSize(ctx, volumeID, newSizeGiB)
-		} else {
-			return 0, nil
-		}
-	}
-
-	err = c.waitForVolumeSize(ctx, volumeID)
-	if newSizeBytes != 0 {
+	// If the volume modification isn't immediately completed, wait for it to finish
+	state := aws.StringValue(response.VolumeModification.ModificationState)
+	if !volumeModificationDone(state) {
+		err = c.waitForVolumeModification(ctx, volumeID)
 		if err != nil {
-			return oldSizeGiB, err
+			return 0, err
 		}
-		return c.checkDesiredSize(ctx, volumeID, newSizeGiB)
-	} else {
-		return 0, c.waitForVolumeSize(ctx, volumeID)
 	}
+
+	// Perform one final check on the volume
+	return c.checkDesiredState(ctx, volumeID, newSizeGiB, options)
```

Comment:
Dumb Q: Is the extra trip to EC2 API worth it? What is being checked here that wasn't checked earlier?

Reply:
If you look at the `checkDesiredState` function, there is a comment above it explaining the potential eventual consistency concerns the previous maintainers implemented it for.

Even if it really isn't necessary, I don't think this PR is the appropriate place to be tearing out such a check.